src/common/options.h

```diff
@@ -39,17 +39,21 @@ struct Option {
    * Advanced: for users, configures some internal behaviour
    * Development: not for users.  May be dangerous, may not be documented.
    */
-  enum level_t {
-    LEVEL_BASIC,
-    LEVEL_ADVANCED,
-    LEVEL_DEV,
+  struct Level {
```

Comment:
if you are interested a syntax like this, please consider http://en.cppreference.com/w/cpp/language/enum#Scoped_enumerations.

Reply:
nice...